pkg/feature/servicemesh/conditions.go

```diff
@@ -107,14 +110,20 @@ func CheckControlPlaneComponentReadiness(ctx context.Context, c client.Client, s
 		Name:      smcpName,
 	}, smcpObj)
 
-	if err != nil {
+	switch {
+	case k8serr.IsNotFound(err):
+		return false, nil
+	case err != nil:
 		return false, fmt.Errorf("failed to find Service Mesh Control Plane: %w", err)
 	}
 
 	components, found, err := unstructured.NestedMap(smcpObj.Object, "status", "readiness", "components")
-	if err != nil || !found {
+	if err != nil {
 		return false, fmt.Errorf("status conditions not found or error in parsing of Service Mesh Control Plane: %w", err)
```

Comment:
```suggestion
		return false, fmt.Errorf("error in parsing of Service Mesh Control Plane: %w", err)
```

Reply:
this has not changed, should we defer it to another PR ?

Reply:
yep,